doc/source/serve/architecture.md

```diff
@@ -24,62 +24,91 @@ There are three kinds of actors that are created to make up a Serve instance:
-deployment or `__call__`) is `async`, the replica will not wait for the
-handler to run; otherwise, the replica will block until the handler returns.
+Each replica maintains a queue of requests and executes requests one at a time, possibly
+using `asyncio` to process them concurrently. If the handler (the deployment function or the `__call__` method of the deployment class) is declared with `async def`, the replica will not wait for the
+handler to run.  Otherwise, the replica will block until the handler returns.
 
-## FAQ
+When making a request via [ServeHandle](serve-handle-explainer) instead of HTTP, the request is placed on a queue in the ServeHandle, and we skip to step 3 above.
 
 (serve-ft-detail)=
 
-### How does Serve handle fault tolerance?
+## Fault tolerance
 
 Application errors like exceptions in your model evaluation code are caught and
 wrapped. A 500 status code will be returned with the traceback information. The
 replica will be able to continue to handle requests.
 
-Machine errors and faults will be handled by Ray. Serve utilizes the [actor
-reconstruction](actor-fault-tolerance) capability. For example, when a machine hosting any of the
-actors crashes, those actors will be automatically restarted on another
+Machine errors and faults will be handled by Ray Serve as follows:
+
+- When replica actors fail, the Controller actor will replace them with new ones.
+- When the HTTP proxy actor fails, the Controller actor will restart it.
+- When the Controller actor fails, Ray will restart it.
+- When using Kuberay ({ref}`kuberay-index`), the Kuberay Operator will recover crashed nodes.
+- When the Ray cluster fails, Ray Serve cannot recover.
+
+When a machine hosting any of the actors crashes, those actors will be automatically restarted on another
 available machine. All data in the Controller (routing policies, deployment
-configurations, etc) is checkpointed to the Ray. Transient data in the
+configurations, etc) is checkpointed to the Ray Global Control Store (GCS) on the head node. Transient data in the
 router and the replica (like network connections and internal request
-queues) will be lost upon failure.
+queues) will be lost for this kind of failure.
+
+(serve-autoscaling-architecture)=
+
+## Ray Serve Autoscaling
+
+Ray Serve's autoscaling feature automatically increases or decreases a deployment's number of replicas based on its load.
+
+![pic](https://raw.githubusercontent.com/ray-project/images/master/docs/serve/autoscaling.svg)
+
+- Each ServeHandle and each replica periodically pushes its metrics to the autoscaler.
+- For each deployment, the autoscaler uses ServeHandle queue metrics and replicas queries metrics to make a decision whether to scale up or down the number of replicas.
+- Each ServeHandle continues to poll the updated group of replicas from the controller. Upon discovery of the new replicas, it will send any buffered or new queries to the replica until `max_concurrent_queries` is reached.
```

Comment:
```suggestion
- Each ServeHandle continuously polls the controller to check for new deployment replicas. Whenever new replicas are discovered, it will send any buffered or new queries to the replica until `max_concurrent_queries` is reached.
```
This sounds like `ServeHandles` briefly stop doing round-robin routing and instead opt to fill up new replicas first before distributing the workload. Is that true?

Reply:
I think that's only true if max_concurrent_queries has been reached on all the existing replicas, which is likely the case if we're autoscaling up.  That's basically a side effect of round robin with the constraint of max_concurrent_queries. 
 I'll update the doc to clarify